packages/components/src/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { PropsWithChildren } from "react"
+import { LayoutAnimation, Pressable, ScaledSize, ViewStyle, useWindowDimensions } from "react-native";
+import { EDSStyleSheet } from "../../styling";
+import { useStyles } from "../../hooks/useStyles";
+import { Paper } from "../Paper";
+import { Portal } from "../Portal";
+
+type DialogProps = PropsWithChildren<{ isOpen: boolean, style?: ViewStyle }> & ({ isDismissable?: false, onClose?: () => void } | { isDismissable: true, onClose: () => void })
+
+export const Dialog = ({ isOpen, style, isDismissable, onClose, children }: DialogProps) => {
+    const dimensions = useWindowDimensions();
+    const styles = useStyles(themeStyles, { style, dimensions })
+    LayoutAnimation.configureNext({ ...LayoutAnimation.Presets.spring, duration: 100 })
+    if (!isOpen) return null;
+    return <Portal name="scrim">
+        <Pressable style={styles.background} onPress={() => isDismissable && onClose()}>
+            <Paper elevation="aboveScrim" style={styles.dialog}>
+                {children}
+            </Paper>
+        </Pressable>
+    </Portal>
+}
+
+const themeStyles = EDSStyleSheet.create((theme, props: { style: ViewStyle | undefined, dimensions: ScaledSize }) => ({
```

Comment:
Instead of passing the style as a prop here, I think it is fine to instead append this style to a style list in the paper prop:

```tsx
<Paper elevation="aboveScrim" style={[styles.dialog, props.style]}>
    {children}
</Paper>
```

Reply:
Wouldn't I need to use StyleSheet.flatten or StyleSheet.compose in order for that to work on the web? Atleast you had to do that before.

Also, I think it's cleaner to leave styling logic outside of the jsx